### PR TITLE
feature/dropout-and-batchnorm

### PR DESCRIPTION
**Pull request by cgarciae**
* Implement `elegy.nn.Dropout` + `elegy.nn.BatchNormalization`
* `mnist_conv.py` example

### CODE REVIEW

File: elegy/nn/batch_normalization.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        self.eps = eps
+        self.scale_init = scale_init or jnp.ones
+        self.offset_init = offset_init or jnp.zeros
+        self.axis = axis
+        self.cross_replica_axis = cross_replica_axis
+        self.channel_index = hk_utils.get_channel_index(data_format)
+        self.mean_ema = hk.ExponentialMovingAverage(momentum, name="mean_ema")
+        self.var_ema = hk.ExponentialMovingAverage(momentum, name="var_ema")
+
+        hk.BatchNorm
```
Comment (sebasarango1180) [inline elegy/nn/batch_normalization.py:114]: @cgarciae sorry for the ignorance:
- What is this void call for? (L114)
- Generally speaking, if there is already an hk.BatchNorm module, why don't just wrap it instead of rewriting it?

**Comment by sebasarango1180** [comment]
I guess the next step would be to implement unit tests for the `nn` submodules. Besides that, the implementation appears to be good to go :+1:

File: elegy/nn/batch_normalization.py (hunk 1 above)
Comment (cgarciae) [reply elegy/nn/batch_normalization.py:114]: 1. Forgot to erase this jejeje
2. Sadly, because we need to inherit from `elegy.Module` which will have logic inside `__call__` to get the summaries.
